src/rpm/headers/header.rs

```diff
@@ -339,7 +339,7 @@ impl Header<IndexSignatureTag> {
     /// Parsing and writing out this section requires knowing how much padding is needed to complete
     /// the alignment.
     pub(crate) fn padding_required(&self) -> u32 {
-        (8 - (self.index_header.data_section_size % 8)) % 8
+        8 - (self.index_header.data_section_size & 0b0111)
```

Comment:
Is this from somewhere or did you do the math yourself?

Reply:
`0b0111` is masking down to the last bits, which is equiv to range `0..7`, which by definition is less than `8` and hence we omit both `modulo` ops.

Reply:
You're absolutely 100% sure it's equivalent?  Because I once made a change that I thought was equivalent, and it was not, and the error looked a lot like the one now showing up in some of the integration tests. 

```
[stdout] /out/full_rpm_sig_protected.rpm:
[stdout] error: /out/full_rpm_sig_protected.rpm: hdr magic: BAD
```

Reply:
Let's push this to a separate PR with unit tests if you want to continue with it

Reply:
Looks like you made the same mistake as me btw.  Instead of `8 - (0...7)` it should be `8 - (1...8)`.  The first case leaves open the possibility of adding 8 bytes of padding which is not what you want.

Reply:
I did a revert, it's such a triviality of a stumbling block